datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -883,7 +893,7 @@ fn both_numeric_or_null_and_numeric(lhs_type: &DataType, rhs_type: &DataType) ->
 ///
 /// Not all operators support dictionaries, if `preserve_dictionaries` is true
 /// dictionaries will be preserved if possible
-fn dictionary_coercion(
+fn dictionary_comparison_coercion(
     lhs_type: &DataType,
     rhs_type: &DataType,
     preserve_dictionaries: bool,
```

Comment:
I think ideally `dictionary_comparison_coercion` should not preserve the dictionary (always false), only value type matters. But since it is used in `comparison_coercion`, the name looks good to me

Reply:
In my mind, most single argument functions / operations / coercions would preserve dictionary encoding if possible as that would keep the data encoded in a more efficent form (dictionary) longer

For example `substr(dict_column, 3, 10)` would ideally produce a `Dictionary` as output (after applying the function to the values)

However, most of our functions don't know how to work with Dictionary arrays, so this would be a larger change.